benchmarks/src/main/java/zipkin2/it/ServerIntegratedBenchmark.java

```diff
@@ -0,0 +1,301 @@
+      frontend.withEnv("JAVA_OPTS", zipkinArg);
+    }
+
+    startContainer(backend);
+    startContainer(frontend);
+
+    String prometheusResource = RELEASED_ZIPKIN_VERSION == null
+      ? "prometheus-local.yml" : "prometheus-container.yml";
+    GenericContainer<?> prometheus = new GenericContainer<>("prom/prometheus")
+      .withNetwork(Network.SHARED)
+      .withNetworkAliases("prometheus")
+      .withExposedPorts(9090)
+      .withCopyFileToContainer(
+        MountableFile.forClasspathResource(prometheusResource), "/etc/prometheus/prometheus.yml");
+    startContainer(prometheus);
+
+    GenericContainer<?> grafana = new GenericContainer<>("grafana/grafana")
+      .withNetwork(Network.SHARED)
+      .withNetworkAliases("grafana")
+      .withExposedPorts(3000)
+      .withEnv("GF_AUTH_ANONYMOUS_ENABLED", "true")
+      .withEnv("GF_AUTH_ANONYMOUS_ORG_ROLE", "Admin");
+    startContainer(grafana);
+
+    startContainer(new GenericContainer<>("appropriate/curl")
+      .withNetwork(Network.SHARED)
+      .withCommand("/create.sh")
+      .withCopyFileToContainer(
+        MountableFile.forClasspathResource("create-datasource-and-dashboard.sh"), "/create.sh"));
+
+    GenericContainer<?> wrk = new GenericContainer<>("skandyla/wrk")
+      .withNetwork(Network.SHARED)
+      .withCommand("-t4 -c128 -d100s http://frontend:8081 --latency");
+    startContainer(wrk);
+
+    LOGGER.info("Benchmark started.");
+    if (zipkin != null) {
+      printContainerMapping(zipkin);
+    }
+    if (storage != null) {{
+      printContainerMapping(storage);
+    }}
+    printContainerMapping(backend);
+    printContainerMapping(frontend);
+    printContainerMapping(prometheus);
+    printContainerMapping(grafana);
+
+    while (wrk.isRunning()) {
+      Thread.sleep(1000);
+    }
+
+    // Wait for prometheus to do a final scrape.
+    Thread.sleep(5000);
+
+    LOGGER.info("Benchmark complete, wrk output:\n{}", wrk.getLogs());
+
+    HttpClient prometheusClient = HttpClient.of(
+      "h1c://" + prometheus.getContainerIpAddress() + ":" + prometheus.getFirstMappedPort());
+
+    LOGGER.info("Messages received: {}", prometheusValue(
+      prometheusClient, "sum(zipkin_collector_messages_total)"));
+    LOGGER.info("Spans received: {}", prometheusValue(
+      prometheusClient, "sum(zipkin_collector_spans_total)"));
+    LOGGER.info("Spans dropped: {}", prometheusValue(
+      prometheusClient, "sum(zipkin_collector_spans_dropped_total)"));
+
+    LOGGER.info("Memory quantiles:");
+    printQuartiles(prometheusClient, "jvm_memory_used_bytes{area=\"heap\"}");
+    printQuartiles(prometheusClient, "jvm_memory_used_bytes{area=\"nonheap\"}");
+
+    LOGGER.info("Total GC time (s): {}",
+      prometheusValue(prometheusClient, "sum(jvm_gc_pause_seconds_sum)"));
+    LOGGER.info("Number of GCs: {}",
+      prometheusValue(prometheusClient, "sum(jvm_gc_pause_seconds_count)"));
+
+    LOGGER.info("POST Spans latency (s)");
```

Comment:
is the logger helping us? seems to add a bunch of redundant formatting :)

Reply:
That's true - switched to println